raiden/log_config.py

```diff
@@ -119,36 +141,94 @@ def add_greenlet_name(
-                repl = "<redacted>"
-            msg = regex.sub(repl, msg)
-        return msg
+        yield f'  File "{filename}", line {lineno}, in {name}\n' f"{line}\n"
 
-    return processor_wrapper
 
+def format_exception_chain(
+    exc_value: BaseException, exc_traceback: Optional[TracebackType], seen: Set[int]
+) -> Iterator[str]:
+    """Recursively format the chain of exceptions for logging.
 
-def _wrap_tracebackexception_format(redact: Callable[[str], str]):
-    """Monkey-patch TracebackException.format to redact printed lines.
+    Notes:
 
-    Only the last call will be effective. Consecutive calls will overwrite the
-    previous monkey patches.
+    - This does not include the stack locals for compactness;
+    - It uses the standard library `linecache` module, but it does not force a
+      cache reset, which saves a few system calls and won't may not show teh
+      wrong stack traces if the files are being edited;
+    - It does not handle exceptions traces larger than the stack depth;
+    - It does not handle SyntaxError specially.
     """
-    original_format = getattr(TracebackException, "_original", None)
-    if original_format is None:
-        original_format = TracebackException.format
-        setattr(TracebackException, "_original", original_format)  # noqa: B010
+    # To avoid infinite recursion each exception is processed only once
+    seen.add(id(exc_value))
 
-    @wraps(original_format)
-    def tracebackexception_format(self, *, chain=True):
-        for line in original_format(self, chain=chain):
-            yield redact(line)
+    cause: Optional[BaseException] = exc_value.__cause__
+    context: Optional[BaseException] = exc_value.__context__
+    suppress_context: bool = exc_value.__suppress_context__
 
-    setattr(TracebackException, "format", tracebackexception_format)  # noqa: B010
+    if cause is not None and id(cause) not in seen:
+        yield from format_exception_chain(cause, cause.__traceback__, seen)
+        yield "\nThe above exception was the direct cause of the following exception:\n\n"
+
+    if context is not None and id(context) not in seen and not suppress_context:
+        yield from format_exception_chain(context, context.__traceback__, seen)
+        yield "\nDuring handling of the above exception, another exception occurred:\n\n"
+
+    if exc_traceback is not None:
+        yield "Traceback (most recent call last):\n"
+        yield from format_traceback(exc_traceback)
+
+    exc_type = type(exc_value)
+
+    # The standard library handles all exceptions ... this is problematic for
+    # our codebase becase it can handle GreenletExits while formatting
+    # exceptions, so instead of using `BaseException` this is using `Exception`
+    try:
+        yield f"{exc_type.__qualname__}: {exc_value}"
+    except Exception:  # Formatting the exception itself can raise an exception
+        yield f"{exc_type.__qualname__}"
+
+
+def format_exception_and_redact_secrets(
```

Comment:
I can understand why you want to switch from a monkey patch to a structlog processor. But isn't this code also a bit complicated? These two functions `format_exception_and_redact_secrets` and `format_exception_chain` are not very easy to read and especially the second seems to try to recreate the way the python standard library does the tracebacks.

At the least we would need a few unit tests for it? I can read the code and understand what it is supposed to do but I can imagine it having bugs since I can't be sure it works just by running it with my mind's eye.